packages/core/src/connectors/types.ts

```diff
@@ -10,7 +15,8 @@ export interface ConnectorMetadata {
 }
 
 // The name `Connector` is used for database, use `ConnectorInstance` to avoid confusing.
-export type ConnectorInstance = EmailConector | SocialConector;
+export type ConnectorInstance = AllConnectors & { connector: Connector };
+export type AllConnectors = EmailConector | SocialConector;
```

Comment:
put `{ connector: Connector }` into BaseConnector? `connector` can be optional

Reply:
good call